src/CreateStudentForm/CreateStudentForm.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
+import './CreateStudentForm.scss'
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { addStudent } from "../actions";
+
+const CreateStudentForm = ({ completeForm, addStudent }) => {
+  const [studentFirstName, setStudentFirstName] = useState('')
+  const [studentLastName, setStudentLastName] = useState('')
+  
+  const isEnabled = 
+    studentFirstName.trim() === '' || 
+    studentLastName.trim() === ''
+  
+  const teacherId = 1
+
+  const postStudent = () => {
+    const url = `https://gumberoo-backend.herokuapp.com/api/v1/teachers/${teacherId}/students/`  
+    return fetch(url, {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({
+        id: teacherId,
+        students: [
+          {
+            first_name: studentFirstName,
+            last_name: studentLastName,
+            age: null
+          }
+        ]
+      })
+    })
+    .then(res => console.log(res))
+    .catch(err => console.error(err))
+  }
+  
+  const createStudent = (e) => {
+    e.preventDefault()
+    postStudent()
+    addStudent({
+      first_name: studentFirstName,
+      last_name: studentLastName,
+      age: null
+    })
+    completeForm()
+  }
+
+  return (
+    <form className='create-student-form' onSubmit={(e) => createStudent(e)}>
+      <h1>Add a Student to Your Roster</h1>
+      <section>
+        <label className='student-name-label' htmlFor='student-firstname-input'>First Name:</label>
+        <input
+          id='student-firstname-input' 
+          className='student-name-input'
+          type='text'
+          placeholder='Enter Student First Name...'
+          value={studentFirstName}
+          onChange={(e) => setStudentFirstName(e.target.value)}
+        />
```

Comment:
This is great - making every student a click - totally stealing this for the lessons.